src/hotspot/share/opto/vectorIntrinsics.cpp

```diff
@@ -2493,13 +2494,18 @@ bool LibraryCallKit::inline_vector_convert() {
     if (is_mask && is_floating_point_type(elem_bt_from)) {
       new_elem_bt_from = elem_bt_from == T_FLOAT ? T_INT : T_LONG;
     }
+    if (is_mask && is_floating_point_type(elem_bt_to)) {
+      new_elem_bt_to = elem_bt_to == T_FLOAT ? T_INT : T_LONG;
```

Comment:
Why we need this change?

Reply:
Before `VectorCast` is used for vector mask casting, the src floating point type has been casted to an integral type that has the same element size. Please see the codes here: https://github.com/openjdk/jdk/blob/master/src/hotspot/share/opto/vectornode.cpp#L1741. We'd better to check the right type of the generated vector cast op.

Reply:
OK, seems fine to me.
Thanks.